scripts/output/extra/disaggregation.R

```diff
@@ -89,15 +88,36 @@ if (length(map_file) > 1) {
 }
 
 .dissagLandConsv <- function(gdx, cfg, map_file, wdpa_hr_file, consv_prio_hr_file) {
-  land_consv_lr <- readGDX(gdx, "pm_land_conservation", react = "silent")
+  land_consv_lr <- readGDX(gdx, "p22_conservation_area", react = "silent")
   wdpa_hr <- read.magpie(wdpa_hr_file)
   map <- readRDS(map_file)
 
   # create full time series
-  land_consv_hr <- new.magpie(map[, "cell"], getYears(land_consv_lr), getNames(wdpa_hr),
-                             sets = c("x.y.iso", "year", getSets(wdpa_hr, fulldim = FALSE)[3]))
-  land_consv_hr[, getYears(land_consv_hr), ] <- wdpa_hr[, nyears(wdpa_hr), ]
-  land_consv_hr[, getYears(wdpa_hr), ] <- wdpa_hr
+  land_consv_hr <- new.magpie(map[, "cell"], getYears(land_consv_lr), getItems(wdpa_hr, dim = 3.2),
+    fill = 0, sets = c("x.y.iso", "year", "data")
+  )
+
+  iso <- readGDX(gdx, "iso")
+  consv_iso <- readGDX(gdx, "policy_countries22")
+  consv_iso <- consv_iso[consv_iso %in% getItems(wdpa_hr, dim = 1.3)]
+  if (length(consv_iso) == 0) {
+    warning("No countries selected in land conservation disaggregation. Results may be erroneous")
+  }
+
+  if (!all(c(cfg$gms$c22_base_protect, cfg$gms$c22_base_protect_noselect) %in% "none")) {
+    base_protect_select <- cfg$gms$c22_base_protect
+    base_protect_noselect <- cfg$gms$c22_base_protect_noselect
+
+    if (base_protect_noselect != "none") {
+      land_consv_hr[, getYears(land_consv_hr), ] <- collapseDim(wdpa_hr[, nyears(wdpa_hr), base_protect_noselect], dim = 3.1)
+      land_consv_hr[, getYears(wdpa_hr), ] <- collapseDim(wdpa_hr[, , base_protect_noselect], dim = 3.1)
+    }
+    if (base_protect_select != "none") {
+      land_consv_hr[consv_iso, , ] <- collapseDim(wdpa_hr[consv_iso, nyears(wdpa_hr), base_protect_select], dim = 3.1)
+    } else if (base_protect_select == "none") {
```

Comment:
This `if` is redundant, isn't it?
```suggestion
    } else {
```

Reply:
This `else` would not be reached if the check in line 107 is FALSE, so that check actually can change things. Is that intended?